hudhook/src/hooks/common.rs

```diff
@@ -116,5 +122,52 @@ pub(crate) fn imgui_wnd_proc_impl(
     let wnd_proc = imgui_renderer.wnd_proc();
     drop(imgui_renderer);
 
+    if imgui_render_loop.as_ref().should_block_messages() {
```

Comment:
I believe we will need to pass in the IO or UI here to be able to correctly determine if we should block the message. The use case is blocking it when the UI wants IO.

Reply:
We _could_ have `Io` here, but not `Ui` as that object only lives during the invocation of `Present`. I don't have a strong opinion here; if we can pass in `ctx.io()` (immutable reference preferably) that would probably be less state management work on the clients, but for use cases that aren't 100% relying on that, clients would have to manage the state anyway. I'd say let's give this a try and change the function to accept `&Io`.